examples/openssl_engine_example.cpp

```diff
@@ -0,0 +1,311 @@
+     */
+    RdKafka::Conf *conf = RdKafka::Conf::create(RdKafka::Conf::CONF_GLOBAL);
+    std::string engine_id;
+    std::string engine_callback_data;
+    int opt;
+
+    while ((opt = getopt(argc, argv, "b:p:c:t:d:i:e:X:")) != -1) {
+        switch (opt) {
+        case 'b':
+            brokers = optarg;
+            break;
+        case 'p':
+            engine_path = optarg;
+            break;
+        case 'c':
+            ca_location = optarg;
+            break;
+        case 'i':
+            engine_id = optarg;
+            break;
+        case 'e':
+            engine_callback_data = optarg;
+            break;
+        case 'd':
+            if (conf->set("debug", optarg, errstr) != RdKafka::Conf::CONF_OK) {
+                std::cerr << errstr << std::endl;
+                exit(1);
+            }
+            break;
+        case 'X': {
+            char *name, *val;
+
+            name = optarg;
+            if (!(val = strchr(name, '='))) {
+                std::cerr << "%% Expected -X property=value, not " <<
+                    name << std::endl;
+                exit(1);
+            }
+
+            *val = '\0';
+            val++;
+
+            if (conf->set(name, val, errstr) != RdKafka::Conf::CONF_OK) {
+                std::cerr << errstr << std::endl;
+                exit(1);
+            }
+        }
+        break;
+
+        default:
+            goto usage;
+        }
+    }
+
+    if (brokers.empty() || engine_path.empty() || optind != argc) {
+    usage:
+        std::string features;
+        conf->get("builtin.features", features);
+        fprintf(stderr,
+            "Usage: %s [options] -b <brokers> -p <engine-path> \n"
+            "\n"
+            "OpenSSL engine integration example. This example fetches metadata\n"
+            "over SSL channel with broker, established using OpenSSL engine.\n"
+            "\n"
+            "librdkafka version %s (0x%08x, builtin.features \"%s\")\n"
+            "\n"
+            " Options:\n"
+            "  -b <brokers>              Broker address\n"
+            "  -p <engine-path>          Path to OpenSSL engine\n"
+            "  -i <engine-id>            OpenSSL engine id\n"
+            "  -e <engine-callback-data> OpenSSL engine callback_data\n"
+            "  -c <ca-cert-location>     File path to ca cert\n"
+            "  -d [facs..]               Enable debugging contexts: %s\n"
+            "  -X <prop=name>            Set arbitrary librdkafka configuration"
+            " property\n"
+            "\n",
+            argv[0],
+            RdKafka::version_str().c_str(), RdKafka::version(),
+            features.c_str(),
+            RdKafka::get_debug_contexts().c_str());
+        exit(1);
+    }
+
+    if (conf->set("bootstrap.servers", brokers, errstr) != 
+        RdKafka::Conf::CONF_OK) {
+        std::cerr << errstr << std::endl;
+        exit(1);
+    }
+
+    if (conf->set("security.protocol", "ssl", errstr) != RdKafka::Conf::CONF_OK) {
```

Comment:
Suggest setting this before the getopt loop so that people can use other security protocols (sasl_ssl)